pkg/safeguards/preprocessing/preprocessing.go

```diff
@@ -0,0 +1,143 @@
+	loadedCharts[mainChartPath] = mainChart
+
+	// Load subcharts and dependencies
+	for _, dep := range mainChart.Metadata.Dependencies {
+		// Resolve the chart path based on the main chart's directory
+		chartPath := filepath.Join(mainChartPath, dep.Repository[len("file://"):])
+		chartPath = filepath.Clean(chartPath)
+
+		subChart, err := loader.Load(chartPath)
+		if err != nil {
+			return nil, fmt.Errorf("failed to load chart: %s", err)
+		}
+		loadedCharts[chartPath] = subChart
+	}
+
+	var manifestFiles []safeguards.ManifestFile
+	for chartPath, chart := range loadedCharts {
+		valuesPath := filepath.Join(chartPath, "values.yaml") // Enforce that values.yaml must be at same level as Chart.yaml
+		mergedValues, err := getValues(chart, valuesPath)
+		if err != nil {
+			return nil, fmt.Errorf("failed to load values: %s", err)
+		}
+		e := engine.Engine{Strict: true}
+		renderedFiles, err := e.Render(chart, mergedValues)
+		if err != nil {
+			return nil, fmt.Errorf("failed to render chart: %s", err)
+		}
+
+		// Write each rendered file to the output directory with the same name as in templates/
+		for renderedPath, content := range renderedFiles {
+			outputFilePath := filepath.Join(tempDir, filepath.Base(renderedPath))
+			if err := os.WriteFile(outputFilePath, []byte(content), 0644); err != nil {
+				return nil, fmt.Errorf("failed to write manifest file: %s", err)
+			}
+			manifestFiles = append(manifestFiles, safeguards.ManifestFile{Name: filepath.Base(renderedPath), Path: outputFilePath})
+		}
+	}
+
+	return manifestFiles, nil
+}
+
+// CreateTempDir creates a temporary directory on the user's file system for rendering templates
+func CreateTempDir(p string) string {
+	dir, err := os.MkdirTemp(p, "prefix")
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	return dir
+}
+
+// IsKustomize checks whether a given path should be treated as a kustomize project
+func IsKustomize(p string) bool {
+	var err error
+	if safeguards.IsYAML(p) {
+		return strings.Contains(p, "kustomization.yaml")
+	} else if _, err = os.Stat(filepath.Join(p, "kustomization.yaml")); err == nil {
+		return true
+	} else if _, err = os.Stat(filepath.Join(p, "kustomization.yml")); err == nil {
+		return true
+	}
+	return false
+}
+
+// Given a kustomization manifest file within kustomizationPath, RenderKustomizeManifest will render templates out to tempDir
+func RenderKustomizeManifest(kustomizationPath, tempDir string) ([]safeguards.ManifestFile, error) {
+	log.Debugf("Rendering kustomization.yaml...")
+	if safeguards.IsYAML(kustomizationPath) {
+		kustomizationPath = filepath.Dir(kustomizationPath)
+	}
+
+	options := &krusty.Options{
+		Reorder:           "none",
+		AddManagedbyLabel: true,
+		LoadRestrictions:  types.LoadRestrictionsRootOnly,
+		PluginConfig:      &types.PluginConfig{},
+	}
+	k := krusty.MakeKustomizer(options)
+
+	// Run the build to generate the manifests
+	kustomizeFS := filesys.MakeFsOnDisk()
```

Comment:
where is this FS on disk? Does it get cleaned up?

Reply:
The current implementation (that I will be taking out bc it's in validate.go) puts the tempdir within the given directory of the kustomization.yaml:

```
tempDir = preprocessing.CreateTempDir(vc.manifestPath)
```

A temporary subdirectory is created there and then promptly removed.